include/onnxruntime/core/graph/graph.h

```diff
@@ -1220,7 +1220,10 @@ class Graph {  // NOLINT(clang-analyzer-optin.performance.Padding): preserve exi
 #endif
 
 #if !defined(ORT_MINIMAL_BUILD)
-  /** Gets the GraphProto representation of this Graph only. */
+  /** Gets the GraphProto representation of this Graph only.
+   * Deprecated. Do not use in main code. This does not recurse the subgraphs,
+   * does not populate the initializers of the root graph.
```

Comment:
I don't think this statement is correct. The implementation of this calls ToGraphProtoInternal which iterates all nodes. When calling Node::ToProto that will descend into the subgraphs if an update is required to create the GraphProto for the subgraph attribute in the node, and that attribute contains any nested subgraphs. 

Prior to converting TensorProto instances to use the in-memory marker this returned a valid GraphProto that could be saved and used any time. In the current implementation if there are in-memory TensorProto instances created the returned GraphProto is only valid for the lifetime of the InferenceSession as there's no logic to change in-memory TensorProtos back to being inlined. Isn't that a breaking change so we can't just say it's deprecated?